lib/Default/MysqlProperties.class.php

```diff
@@ -0,0 +1,71 @@
+<?php declare(strict_types=1);
+
+
+use Dotenv\Dotenv;
+
+
+class MysqlProperties {
+	private const CONFIG_MYSQL_HOST = "MYSQL_HOST";
+	private const CONFIG_MYSQL_USER = "MYSQL_USER";
+	private const CONFIG_MYSQL_PASSWORD = "MYSQL_PASSWORD";
+	private const CONFIG_MYSQL_PORT = "MYSQL_PORT";
+	private const CONFIG_MYSQL_DATABASE = "MYSQL_DATABASE";
+	private string $host;
+	private string $user;
+	private string $password;
+	private int $port;
+	private string $databaseName;
+
+	public function __construct() {
+		$configArray = !isset($_ENV["MYSQL_CONFIG_FROM_ENVIRONMENT"])
+			? Dotenv::createArrayBacked(ROOT)->load()
+			: $_ENV;
+		$this->parseConfigArray($configArray);
+	}
+
+	private function parseConfigArray(array $array) {
```

Comment:
That works for me, I think I may have initially had another idea at some point (like testing this), but I suppose I could test it all from the constructor -- it would just be a little more involved than testing a pure, discrete method with a single responsibility.

Reply:
Fixed with new validation stuff